### DISCUSSION TITLE
weird behavior w/ NextAuth SignIn and redirect callbackURL

### DISCUSSION THREAD

**Discussion by vincanger** [rainbow-me/rainbowkit#729] [Q&A]
I'm using RainbowKit with SIWE and NextAuth. I want to redirect on a successful signature verification, and I noticed some weird behavior if I use the proposed implementation, like this:

```js
const authAdapter = createAuthenticationAdapter({
  //...

    verify: async ({ message, signature }) => {
      const response = await signIn('credentials', {
        message: JSON.stringify(message),
        signature,
        redirect: true,
        callbackUrl: window.location.origin + '/rainbow-protected',
      });
      return response?.ok ?? false;
    },

 //...
```

like so, it will flash an error message for a second before successfully signing in and redirecting:

<img width="80%" src="https://i.imgur.com/4XjqrKB.png"/>

I noticed that this is because when `redirect = true`, the `signIn` function from NextAuth no longer returns anything -- it used to return a `Promise<Boolean>`. 

I settled on this implementation by simply returning `true` in all cases, because if the signature verification fails, it won't redirect anyway:

```js
    verify: ({ message, signature }) => {
      signIn('credentials', {
        message: JSON.stringify(message),
        signature,
        redirect: true,
        callbackUrl: window.location.origin + '/rainbow-protected',
      });
      return new Promise((resolve, reject) => {
        resolve(true);
      });
    },
```

wondering if there's a better way to do this or if this will cause any unexpected behavior? I logged the return value of `signIn` when`redirect = true`, and it's just an empty object. Looks like `signIn` is no longer a Promise when redirecting, and currently the `AuthenticationAdapter` interface for RainbowKit has the `verify` method always returning a Promise:

```js
    verify: (args: {
        message: Message;
        signature: string;
    }) => Promise<boolean>;
```

**Comment by joakim-roos** [comment]
Got the same problem here. Did returning a promise cause any issues for you? Otherwise i'll give that a try

**Reply by vincanger** [reply]
I didn't notice any issues yet :)

because `signIn` is now synchronous, it will throw an error if something goes wrong before it reaches the promise

**Comment by acoutts** [comment]
I found the same behavior with web3modal and siwe / nextauth - just the same as you here. Thanks for your fix, it seems to work for me too.
